src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/OffenderServiceTest.kt

```diff
@@ -310,115 +312,152 @@ class OffenderServiceTest {
+    }
 
-    every { mockApDeliusContextApiClient.getUserAccessForCrns("distinguished.name", listOf(crn)) } returns ClientResult.Success(HttpStatus.OK, UserAccess(listOf(caseAccess)))
+    @Test
+    fun `return error if greater than 500 crns are requested`() {
+      assertThatThrownBy {
+        offenderService.canAccessOffenders("distinguished.name", (0..500).map { "$it" })
+      }.isInstanceOf(InternalServerErrorProblem::class.java)
+        .hasMessage("Internal Server Error: Cannot bulk request access details for more than 500 CRNs. 501 have been provided.")
+    }
 
-    val result = offenderService.canAccessOffenders("distinguished.name", listOf(crn))
-    assertThat(result[crn]).isFalse()
-  }
+    @Test
+    fun `return false when crn is user excluded from viewing`() {
+      val crn = randomNumberChars(8)
+      val caseAccess = CaseAccessFactory()
+        .withCrn(crn)
+        .withUserExcluded(true)
+        .produce()
 
-  @Test
-  fun `getUserAccessForOffenderCrns return false when crn is user restricted from viewing`() {
-    val crn = randomNumberChars(8)
-    val caseAccess = CaseAccessFactory()
-      .withCrn(crn)
-      .withUserRestricted(true)
-      .produce()
+      every {
+        mockApDeliusContextApiClient.getUserAccessForCrns(
+          "distinguished.name",
+          listOf(crn),
+        )
+      } returns ClientResult.Success(HttpStatus.OK, UserAccess(listOf(caseAccess)))
 
-    every { mockApDeliusContextApiClient.getUserAccessForCrns("distinguished.name", listOf(crn)) } returns ClientResult.Success(HttpStatus.OK, UserAccess(listOf(caseAccess)))
+      val result = offenderService.canAccessOffenders("distinguished.name", listOf(crn))
+      assertThat(result[crn]).isFalse()
+    }
 
-    val result = offenderService.canAccessOffenders("distinguished.name", listOf(crn))
-    assertThat(result[crn]).isFalse()
-  }
+    @Test
+    fun `return false when crn is user restricted from viewing`() {
+      val crn = randomNumberChars(8)
+      val caseAccess = CaseAccessFactory()
+        .withCrn(crn)
+        .withUserRestricted(true)
+        .produce()
 
-  @Test
-  fun `getUserAccessForOffenderCrns return false when crn not have access result`() {
-    val crn = randomNumberChars(8)
+      every {
+        mockApDeliusContextApiClient.getUserAccessForCrns(
+          "distinguished.name",
+          listOf(crn),
+        )
+      } returns ClientResult.Success(HttpStatus.OK, UserAccess(listOf(caseAccess)))
 
-    every { mockApDeliusContextApiClient.getUserAccessForCrns("distinguished.name", listOf(crn)) } returns ClientResult.Success(HttpStatus.OK, UserAccess(emptyList()))
+      val result = offenderService.canAccessOffenders("distinguished.name", listOf(crn))
+      assertThat(result[crn]).isFalse()
+    }
 
-    val result = offenderService.canAccessOffenders("distinguished.name", listOf(crn))
-    assertThat(result[crn]).isFalse()
-  }
+    @Test
+    fun `return false when crn not have access result`() {
```

Comment:
good shout! I changed it to `return false when no access result returned for crn`. i'm not actually sure this would ever happen but good to guard against it